crates/bins/src/bin/datadog_static_analyzer_server/cli.rs

```diff
@@ -164,20 +174,26 @@ pub fn prepare_rocket(tx_keep_alive_error: Sender<i32>) -> Result<RocketPreparat
             log_dir,
             format!("server.{}.log", std::process::id()),
         )?;
-        let (non_blocking, guard) = tracing_appender::non_blocking(file_appender);
-
-        tracing_subscriber::fmt()
-            .with_env_filter(EnvFilter::from_default_env())
-            .json()
-            .with_writer(non_blocking)
-            .init();
-        Some(guard)
+        let (non_blocking_file, file_guard) = tracing_appender::non_blocking(file_appender);
+
+        let file_layer = tracing_subscriber::fmt::layer()
+            .with_writer(non_blocking_file)
+            .json();
+
+        tracing_subscriber::registry()
```

Comment:
NIT: you could set the common layers outside the `if let` scope so you wouldn't have to repeat them.

Reply:
For instance: 
```rust
    let builder = tracing_subscriber::registry()
        .with(EnvFilter::from_default_env())
        .with(stdouterr_layer);

// and then in the `if let` branches:
// ... if
builder.with(file_layer).init();
// ... else
builder.init();
        
```